src/main/java/service/PetServiceMain.java

```diff
@@ -0,0 +1,49 @@
+package service;
+import model.Pet;
+
+import java.util.Map;
+public class PetServiceMain implements PetService{
+    private final Map<String, Pet> pets;
+
+    public PetServiceMain(Map<String, Pet> pets) {
+        this.pets = pets;
+    }
+
+    @Override
+    public void add(Pet pet) {
+        try {
+            if (pets != null) {
+                String key = Integer.toString(pets.size() + 1);
+                pets.put(key, pet);
+            }
+        } catch (RuntimeException e) {
+            throw new IllegalArgumentException("Pet could not be added");
+        }
+    }
+
+    @Override
+    public void delete(String key) {
+        try {
+            if (pets != null) {
+                pets.remove(key);
+            }
+        } catch (RuntimeException e) {
+            throw new IllegalArgumentException("Pet with key " + key + " could be delete.");
+        }
+    }
+
```

Comment:
error handling is bad, removing value from the map should not lead to runtime exception

Reply:
need to be redesigned